ecl/agentexec/agentexec.cpp

```diff
@@ -284,14 +290,23 @@ class WaitThread : public CInterfaceOf<IPooledThread>
                     std::list<std::pair<std::string, std::string>> params = { };
                     if (compConfig->getPropBool("@useThorQueue", true))
                         params.push_back({ "queue", queue.get() });
-                    StringBuffer jobName(wuid);
                     if (isThorAgent)
                     {
-                        jobName.append('-').append(graphName);
                         params.push_back({ "graphName", graphName.get() });
                         params.push_back({ "wfid", std::to_string(wfid) });
                     }
 
+                    const char *targetName = compConfig->queryProp("@targetName");
+                    StringBuffer jobName;
+                    if (targetName)
+                    {
+                        jobName.append(targetName).append('-');
+                        jobName.append(myInstanceNumber);
+                    }
+                    else
+                        jobName.append(wuid).append('-').append(graphName);
+                    params.push_back({ "instanceNum", std::to_string(myInstanceNumber) });
```

Comment:
should move this into the targetName block, if there's no targetName, then there instanceNum won't be usable.